src/dex.hs

```diff
@@ -46,7 +67,7 @@ runMode evalMode preludeFile opts = do
   let runEnv m = evalStateT m env
   case evalMode of
     ReplMode prompt ->
-      runEnv $ runInputT defaultSettings $ forever (replLoop prompt opts)
+      runEnv $ runInputT hasklineSettings $ forever (replLoop prompt opts)
```

Comment:
AFAICT all the things we need to have completion of global function names etc is available to us in the `env` monad.
But I have no real idea how to get it out of there and into the monad that `hasklineSettings`/`keywordCompletions` gets.